Coverage/FileInfo.h

```diff
@@ -3,48 +3,112 @@
+	static constexpr std::string_view FORWARD_SLASH_ASTERISK_LINE = "/*";
+	static constexpr std::string_view FORWARD_SLASH_ASTERISK_LINE_END = "*/";
 
-		std::string line;
-		while (std::getline(ifs, line))
+	bool IsCoverageFlag(const std::string::const_iterator& iter, const ptrdiff_t iterSize,
+											const std::string_view& coverageFlag, const bool multilineComment)
+	{
+		if (iterSize < coverageFlag.size())
 		{
-			// Process str
-			size_t idx = line.find("#pragma");
+			return false;
+		}
+		if (iterSize == coverageFlag.size())
+		{
+			return std::equal(coverageFlag.begin(), coverageFlag.end(), iter);
+		}
+
+		if (!multilineComment)
+		{
+			return false;
+		}
+
+		if (!std::equal(coverageFlag.begin(), coverageFlag.end(), iter))
+		{
+			return false;
+		}
+
+		const ptrdiff_t restSize = iterSize - coverageFlag.size();
+		if (restSize < FORWARD_SLASH_ASTERISK_LINE_END.size()) {
+			return false;
+		}
+
+		return std::equal(FORWARD_SLASH_ASTERISK_LINE_END.begin(),
+											FORWARD_SLASH_ASTERISK_LINE_END.end(),
+											iter + coverageFlag.size());
+	}
+
+	LineType GetLineType(const std::string& line)
+	{
+		static const std::vector<std::string_view> prefixCoverage{ PRAGMA_LINE, DOUBLE_FORWARD_SLASH_LINE, FORWARD_SLASH_ASTERISK_LINE };
+		for (const std::string_view& prefix : prefixCoverage)
+		{
+			const size_t idx = line.find(prefix);
 			if (idx != std::string::npos)
 			{
-				size_t jdx = line.find_first_not_of(' ', idx + 7);
-				if (jdx != std::string::npos)
+				std::string::const_iterator jdx = std::find_if_not(line.begin() + idx + prefix.length(), line.end(), std::isspace);
+				if (jdx != line.end())
 				{
-					std::string prag = line.substr(jdx);
-					size_t kdx = prag.find(' ');
-					if (kdx != std::string::npos)
-					{
-						prag = prag.substr(0, kdx);
-					}
+					std::string::const_iterator kdx = std::find_if(jdx, line.end(), std::isspace);
+					const ptrdiff_t size = kdx - jdx;
+					const bool multiLineComment = prefix == FORWARD_SLASH_ASTERISK_LINE;
 
-					if (prag == "DisableCodeCoverage")
+					if (IsCoverageFlag(jdx, size, DISABLE_COVERAGE, multiLineComment))
 					{
-						current = false;
+						return LineType::DISABLE_COVERAGE;
 					}
-					else if (prag == "EnableCodeCoverage")
+					if (IsCoverageFlag(jdx, size, ENABLE_COVERAGE, multiLineComment))
 					{
-						relevant.push_back(current);
-						current = true;
-						continue;
+						return LineType::ENABLE_COVERAGE;
 					}
 				}
 			}
+		}
+		return LineType::CODE;
+	}
+
+public:
+	FileInfo(const std::string& filename)
+	{
+		std::ifstream ifs(filename);
+
+		bool current = true;
+
+		std::string line;
+		while (std::getline(ifs, line))
+		{
+			// Process str
+			LineType lineType = GetLineType(line);
+			if (lineType == LineType::DISABLE_COVERAGE)
+			{
+				current = false;
+			}
+			else if (lineType == LineType::ENABLE_COVERAGE)
+			{
+				relevant.push_back(current);
+				current = true;
+				continue;
```

Comment:
You have to blink a few times with your eyes before understanding what's going on here. Duplicate 113 to line 104 and 
an 'else' and remove the 'continue' for clarity please.